app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -8,36 +8,61 @@ import androidx.lifecycle.ViewModel
 import org.oppia.android.R
 import org.oppia.android.app.home.HomeItemViewModel
 import org.oppia.android.app.home.RouteToRecentlyPlayedListener
+import org.oppia.android.app.model.RecommendedActivityList
 
 /** [ViewModel] for the promoted story list displayed in [HomeFragment]. */
 class PromotedStoryListViewModel(
   private val activity: AppCompatActivity,
-  val promotedStoryList: List<PromotedStoryViewModel>
+  val promotedStoryList: List<PromotedStoryViewModel>,
+  private val recommendedActivityList: RecommendedActivityList
 ) : HomeItemViewModel() {
   private val routeToRecentlyPlayedListener = activity as RouteToRecentlyPlayedListener
 
-  // TODO(#2297): Update this span count and move to values/integers.xml once behavior is clarified
-  private val promotedStoriesTabletSpanCount: Int =
-    if (Resources.getSystem().configuration.orientation == Configuration.ORIENTATION_PORTRAIT) 2
-    else 3
-
   /** Returns the padding placed at the end of the promoted stories list based on the number of promoted stories. */
   val endPadding =
     if (promotedStoryList.size > 1)
       activity.resources.getDimensionPixelSize(R.dimen.home_padding_end)
     else activity.resources.getDimensionPixelSize(R.dimen.home_padding_start)
 
   /** Determines and returns the visibility for the "View All" button. */
+  fun getHeader(): String {
+    recommendedActivityList.recommendedStoryList.let {
```

Comment:
Oh instead of using let we could `with` as that would be more appropriate here and elsewhere as it is an scope function which would change the scope and then we won't require to write even `it`.

Reply:
@aggarwalpulkit596 can you please suggest me how I can use with here instead of let.

Reply:
```
with(recommendedActivityList.recommendedStoryList) {
      return when {
        suggestedStoryList.isNotEmpty() -> {
          if (recentlyPlayedStoryList.isNotEmpty() || olderPlayedStoryList.isNotEmpty()) {
            activity.getString(R.string.stories_for_you)
          } else
            activity.getString(R.string.recommended_stories)
        }
        recentlyPlayedStoryList.isNotEmpty() -> {
          activity.getString(R.string.recently_played_stories)
        }
        else -> {
          activity.getString(R.string.last_played_stories)
        }
      }
    }
```

Reply:
`with` changes the the current scope to the object so here `this` is referring to `recommendedActivityList.recommendedStoryList` and since is implicitly applied here we don't have to write this here. Does that make sense ?

Reply:
Yes, got it done. Thanks Pulkit! :)